src/Stryker.Core/Stryker.Core/MutationTest/MutationTestProcess.cs

```diff
@@ -38,20 +41,34 @@ public class MutationTestProcess : IMutationTestProcess
         public MutationTestProcess(MutationTestInput mutationTestInput,
             IReporter reporter,
             IEnumerable<IMutator> mutators,
-            IMutationTestExecutor mutationTestExecutor,
-            IMutantOrchestrator orchestrator = null,
+            IMutationTestExecutor mutationTestExecutor, IMutantOrchestrator orchestrator = null,
             ICompilingProcess compilingProcess = null,
-            IFileSystem fileSystem = null)
+            IFileSystem fileSystem = null,
+            IProgressReporter progressReporter = null)
         {
             _input = mutationTestInput;
             _reporter = reporter;
             _mutationTestExecutor = mutationTestExecutor;
+            _progressReporter = progressReporter ?? CreateProgressReporter();
             _orchestrator = orchestrator ?? new MutantOrchestrator(mutators);
             _compilingProcess = compilingProcess ?? new CompilingProcess(mutationTestInput, new RollbackProcess());
             _fileSystem = fileSystem ?? new FileSystem();
             _logger = ApplicationLogging.LoggerFactory.CreateLogger<MutationTestProcess>();
         }
 
+        private static ProgressReporter CreateProgressReporter()
```

Comment:
In my opinion this should be placed inside the reporter factory. This makes (runtime) switching between reporters more easy.